docs/src/pages/customization/unstyled-components/unstyled-components.md

```diff
@@ -0,0 +1,138 @@
+- hooks
+
+Most of the unstyled components are implemented with the help of a hook (where it makes sense). Hooks encapsulate logic, while components provide structure.
+
+When creating custom components based on the unstyled ones, you can use either unstyled components or hooks.
+
+With components, you will usually be able write less code, but with hooks you have the ultimate control over the structure of the rendered HTML.
+
+## Anatomy of an unstyled component
+
+While each component has its own API, there are a few props common to all of them:
+
+- `components` - an object that allows you to override subcomponents used by the unstyled (host) component. Each host component will at least have the `Root` slot. Many complex components have more slots. You can either provide a custom component or an HTML tag there.
+
+  ```jsx
+  <BadgeUnstyled components={{ Root: 'div', Badge: MyCustomBadge }} />
+  ```
+
+- `component` - a shortcut to `components.Root`. Note that if both `components.Root` and `component` are specified, `component` takes precedence.
+
+  ```jsx
+  <BadgeUnstyled component="div" />
+  ```
+
+- `componentsProps` - an object with each slot's props. Even if you don't override a specific slot (with `components` or `component`), you can provide additional props for the default components. For example, if you want to add a custom CSS class to one of the slots' components, specify it in the `componentsProps`.
+
+  ```jsx
+  <BadgeUnstyled componentsProps={{ input: { className: 'my-badge' } }} />
+  ```
+
+Additionally, all the extra props placed on the host component are propagated into the root component (just as if they were placed in `componentsProps.root`). These two examples are equivalent:
+
+```jsx
+<BadgeUnstyled id="badge1">
+```
+
+```jsx
+<BadgeUnstyled componentsProps={{ root: { id: 'badge1' } }}>
+```
+
+## Styling the unstyled
+
+There are several levels of customization available:
+
+### Applying custom CSS rules to unstyled components
+
+If you're happy with the structure of the HTML rendered by the unstyled component by default, you can just apply custom styles to the classes used by the component. Each component has its own set of classes. Some are static - present on different parts of the component all the time, and some are applied conditionally (like `Mui-disabled`, applied when a component is disabled).
+
+Each component's API documentation lists all classes that the component uses. Additionally, you can import a `[componentName]Classes` object that describes all the classes a given component uses (see the demo below for an example).
+
+{{"demo": "pages/customization/unstyled-components/styling-customCss.js"}}
+
+### Overriding the unstyled components' slots
+
+If you don't want to rely on default components, you can override them with your own. This makes it possible to provide a styled component. Each unstyled component has a specific set of "slots" - that is subcomponents that you can override.
```

Comment:
We should use the same words as above. For example, use slots everywhere, or use components everywhere.